code/iec61131/src/main/kotlin/org/fbme/lib/iec61131/model/Iec61131Xml.kt

```diff
@@ -0,0 +1,396 @@
+package org.fbme.lib.iec61131.model
+
+import kotlinx.serialization.*
+import kotlinx.serialization.descriptors.buildClassSerialDescriptor
+import kotlinx.serialization.encoding.Decoder
+import kotlinx.serialization.encoding.Encoder
+import nl.adaptivity.xmlutil.EventType
+import nl.adaptivity.xmlutil.serialization.XML
+import nl.adaptivity.xmlutil.serialization.XmlValue
+import kotlin.io.path.Path
+import kotlin.io.path.readText
+
+class Iec61131Xml {
```

Comment:
I think there is no major benefit of making all serializable classes nested inside a single class. Having them all top-level is fine.